internal/bminventory/inventory_test.go

```diff
@@ -14038,8 +14246,14 @@ var _ = Describe("GetInfraEnvPresignedFileURL", func() {
 			Expect(u.Query().Get("image_token")).To(Equal(""))
 			Expect(u.Query().Get("api_key")).To(Equal(""))
 			Expect(u.Query().Get("file_name")).To(Equal("discovery.ign"))
+			Expect(u.Query().Get("boot_control")).To(BeEmpty())
 			Expect(u.Path).To(Equal(fmt.Sprintf("/api/assisted-install/v2/infra-envs/%s/downloads/files", infraEnvID.String())))
 		})
+
+		It("generates a url with no token for discovery.ign with boot_control", func() {
```

Comment:
This isn't what this test is testing for.
This should be something like `returns bad request when boot_control is used with discovery.ign`